cdap-api/src/main/java/co/cask/cdap/api/schedule/TimeTriggerInfo.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import javax.annotation.Nullable;
+
+/**
+ * The time trigger information to be passed to the triggered program.
+ */
+public class TimeTriggerInfo extends TriggerInfo {
+  private final String cronExpression;
+  @Nullable
+  private String logicalStartTime;
```

Comment:
This is null also because of the OR trigger, in which the TimeTrigger may not be satisfied